gorm/fieldmask.go

```diff
@@ -24,6 +24,7 @@ func MergeWithMask(source, dest interface{}, mask *fieldmask.FieldMask) error {
 	if reflect.TypeOf(source) != reflect.TypeOf(dest) {
 		return errors.New("Types of source and destination objects do not match")
 	}
+pathsloop:
```

Comment:
what kind of black magic is this?

Reply:
https://www.ardanlabs.com/blog/2013/11/label-breaks-in-go.html

Reply:
I'd say to use labels sparingly -- seeing a label in code sets off my abuse-of-control-flow alarms in general. I'm ok with it here, but please let's not do it everywhere :)

Reply:
I guess I'll rethink my plan of refactoring everything to use labels and `goto`s instead of function calls then...

Really though, I think this is... either the first or second time I've ever used one. I'm quite sparing.